modules/siddhi-extensions/markov-models/src/main/java/org/wso2/siddhi/extension/markov/MarkovChainTransitionProbabilitiesCalculator.java

```diff
@@ -216,22 +223,37 @@ private void populateMarkovMatrix(String markovMatrixStorageLocation) {
         } catch (IOException e) {
             log.error("Exception occurred while reading the data file " + markovMatrixStorageLocation, e);
         } finally {
-            try {
-                if (fis != null) {
-                    fis.close();
-                }
-                if (bis != null) {
-                    bis.close();
-                }
-                if (br != null) {
-                    br.close();
-                }
-            } catch (IOException ex) {
-                log.error("Exception occurred when closing the file stream of the data file "
-                        + markovMatrixStorageLocation, ex);
-            }
+            closeQuietly(bufferedReader, bufferedInputStream, fileInputStream);
         }
         markovMatrix.setStartStateCount(startStateCount);
         markovMatrix.setTransitionCount(transitionCount);
     }
+
+    private void closeQuietly(BufferedReader bufferedReader, BufferedInputStream bufferedInputStream,
```

Comment:
private void closeQuietly(Closable... closables) {
if(closables != null) {for (Closable closanle : closables) {try{closable.close()} catch() {/*handle*/}}}